src/main/java/io/supertokens/session/accessToken/AccessToken.java

```diff
@@ -414,10 +452,10 @@ static AccessTokenInfo fromJSON(AppIdentifier appIdentifier, JsonObject payload,
                         payload.get("iat").getAsLong() * 1000,
                         version,
                         new TenantIdentifier(appIdentifier.getConnectionUriDomain(), appIdentifier.getAppId(),
-                                payload.get("tId").getAsString())
+                                payload.has("tId") ? payload.get("tId").getAsString() : null)
```

Comment:
you should explicitly check for v3 and if it's v3, set the tenantId to null

Reply:
Done